### PR TITLE
[FIX] stock_account: use “account_journal_id” only if the valuation is real_time

### PR DESCRIPTION
**Pull request by DjamelTouati**
Steps to reproduce the bug:
- Go to inventory > create a new product
    -Create a new product category:
        * Select "Manual" inventory Valuation
        * Choose "First In First Out (FIFO)" costing Method
        * Be sure that in "Automated" inventory valuation, stock journal is not settled
- Save
- Update quantity of the product
- Go to Inventory > Inventory Valuation> Choose the product newly created
- Click on the "+" button to make manual valuation

Problem:
An error is triggered because we try to access stock_journal when it is false

Solution:
Use “account_journal_id” only if the valuation is real_time and relax the required constraint of `account_journal_id` in case of manual inventory Valuation (which is not used in that case)

Opw-2541497

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock_account/wizard/stock_valuation_layer_revaluation.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ def default_get(self, default_fields):
                 raise UserError(_("You cannot revalue a product with a standard cost method."))
             if product.quantity_svl <= 0:
                 raise UserError(_("You cannot revalue a product with an empty or negative stock."))
-            if 'account_journal_id' not in res and 'account_journal_id' in default_fields:
+            if 'account_journal_id' not in res and 'account_journal_id' in default_fields and self.property_valuation == 'real_time':
```
Comment (ryv-odoo) [inline addons/stock_account/wizard/stock_valuation_layer_revaluation.py:24]: `self` isn't instantiate yet, then `self.property_valuation` will be empty (=> use `product.categ_id.property_valuation` instead). 
Also can you move the line `accounts = product.product_tmpl_id.get_product_accounts()` inside the this if ?

File: addons/stock_account/wizard/stock_valuation_layer_revaluation.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def default_get(self, default_fields):
     new_value_by_qty = fields.Monetary("New value by quantity", compute='_compute_new_value')
     reason = fields.Char("Reason", help="Reason of the revaluation")
 
-    account_journal_id = fields.Many2one('account.journal', "Journal", required=True, check_company=True)
+    account_journal_id = fields.Many2one('account.journal', "Journal", check_company=True)
```
Comment (ryv-odoo) [inline addons/stock_account/wizard/stock_valuation_layer_revaluation.py:42]: Don't forget to put the required condition in the views if the `property_valuation == 'real_time'` (via attrs)

**Comment by DjamelTouati** [comment]
Thanks @ryv-odoo for the review, Could you check the update ?
Thanks !

**Review by ryv-odoo — APPROVED** [review]

**Comment by Whenrow** [comment]
robodoo r+

**Pull request closed without merging**